packages/x-components/src/components/result/__tests__/base-result-add-to-cart.spec.ts

```diff
@@ -1,47 +1,86 @@
-import { mount } from '@vue/test-utils';
+import { Result } from '@empathyco/x-types';
+import { mount, Wrapper } from '@vue/test-utils';
 import { getResultsStub } from '../../../__stubs__/results-stubs.factory';
 import { getDataTestSelector, installNewXPlugin } from '../../../__tests__/utils';
+import { XBus } from '../../../plugins/index';
+import { Dictionary } from '../../../utils/index';
 import BaseResultAddToCart from '../base-result-add-to-cart.vue';
 
 describe('testing BaseResultAddToCart component', () => {
   const result = getResultsStub()[0];
-  const [, localVue] = installNewXPlugin();
 
-  it('emits UserClickedResultAddToCart when the user click on the component', () => {
-    const resultAddToCartWrapper = mount(BaseResultAddToCart, {
-      localVue,
-      propsData: { result }
-    });
+  function renderAddToCart({
+    result = { name: 'TEST' } as Result,
+    template = '<BaseResultAddToCart :result="result"/>',
+    methods = {}
+  }: RenderAddToCartOptions): RenderAddToCartApi {
+    const [, localVue] = installNewXPlugin();
+    const wrapper = mount(
+      { template, data: () => ({ result }), methods },
```

Comment:
Instead of using the data, should we pass the result as a prop by using propsData and props?